app/models/claim/matching_attribute_finder.rb

```diff
@@ -41,11 +41,11 @@ def matching_claims
         vals = values_for_attributes(@source_claim.eligibility, attributes)
         next if vals.blank?
 
-        concatenated_columns = "CONCAT(#{attributes.join(",")})"
+        concatenated_columns = "CONCAT(#{attributes.join(",")})" # current policies will only have a single value in attributes
         policies_to_find_matches.map { |policy|
-          policy::Eligibility.where("LOWER(#{concatenated_columns}) = LOWER(?)", vals.join)
+          policy::Eligibility.where("LOWER(#{concatenated_columns}) = LOWER(?)", vals.join) if all_attributes_exist_on_the_eligibility(attributes, policy::Eligibility)
```

Comment:
Why do we need the `all_attributes_exist_on_the_eligibility` check? Would we ever define `Policies::SomePolicy::ELIGIBILITY_MATCHING_ATTRIBUTES` with attributes that aren't present on the policy's eligibility?